Core/automation/lib/python/core/date.py

```diff
@@ -34,87 +54,154 @@ def remove_java_converter(clazz):
-from org.eclipse.smarthome.core.library.types import DateTimeType
+from datetime import datetime as pythonDateTime, date as pythonDate
+from org.joda.time import DateTime as jodaDateTime, DateTimeZone as jodaDateTimeZone
+from java.util import Calendar as utilCalendar, Date as utilDateTime, TimeZone as utilTimeZone
+from org.eclipse.smarthome.core.library.types import DateTimeType as eclipseDateTime
+
+try:
+    # if the compat1.x bundle is not installed the 1.x DateTimeType is not available
+    from org.openhab.core.library.types import DateTimeType as legacyDateTime
+except:
+    pass
 
-__all__ = ["ZonedDateTime", "format_date", 
-           "days_between", "hours_between", "minutes_between", "seconds_between",
-           "to_java_zoneddatetime", "toJTime", "to_java_calendar", "toJCal", 
-           "to_python_datetime", "toPyDT", "pythonTimezone",
-           "to_joda_datetime", "toJodaDT"]
+__all__ = [
+    "format_date",
+    "days_between", "hours_between", "minutes_between", "seconds_between",
+    "to_java_zoneddatetime", "toJTime", "javaDateTime",
+    "to_java_calendar", "toJCal",
+    "to_python_datetime", "toPyDT", "pythonDateTime", "pythonTimezone",
+    "to_joda_datetime", "toJodaDT", "jodaDateTime"
+]
 
 
 def format_date(value, format_string="yyyy-MM-dd'T'HH:mm:ss.SSxx"):
     """Returns string of ``value`` formatted according to ``format_string``.
 
     This function can be used when updating Items in openHAB or to format any
-    DateTime value for output.
+    DateTime value for output. The default format string follows the same
+    ISO8601 format used in openHAB.
 
     Examples:
         .. code-block::
 
-            sendCommand("item_name", date.format_date(date_value))
+            sendCommand("date_item", format_date(date_value))
+            log.info("The time is currently: {}".format(format_date(javaDateTime.now())))
 
     Args:
-        value: any known DateTime value.
-        format_string (string): pattern to format ``value`` with.
+        value: Any known DateTime value.
+        format_string (str): Pattern to format ``value`` with.
             See `java.time.format.DateTimeFormatter <https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html>`_
             for format string tokens.
+
+    Returns:
+        | A string representation of ``value`` according to ``format_string``.
+        | If ``value`` does not have timezone information, the system default
+          will be used.
     """
     return to_java_zoneddatetime(value).format(DateTimeFormatter.ofPattern(format_string))
 
 def days_between(value_from, value_to, calendar_days=False):
-    '''Returns number of whole days between value_from and value_to. Setting 
-    calendar_days=True will provide the number of calendar days, rather than 
-    24 hour intervals.
-    Accepts any date type used by this module'''
+    """Returns the number of days between ``value_from`` and ``value_to``.
```

Comment:
Please update the docstrings to standardize them to a format of...

```
"""
Double quotes on separate lines... before and after
"""
```
IMO, it looks better and is easier to read.